File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        set_char = set()
        left = 0
        len_max = 0

        for right in range(len(s)):
            while s[right] in set_char:
                set_char.remove(s[left])
                left+=1
            set_char.add(s[right])
            len_max = max(len_max, right - left + 1)

        return len_max

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int n = s.length();
        int maxLength = 0;
        
        for(int i = 0; i < n; i++){
            
            HashSet<Character> set = new HashSet<>();
            
            for(int j = i; j < n; j++){
                
                char ch = s.charAt(j);
                
                if(set.contains(ch)){
                    break;
                }
                
                set.add(ch);
                maxLength = Math.max(maxLength, j - i + 1);
            }
        }
        
        return maxLength;
    }
}
